frame/evm/src/runner/stack.rs

```diff
@@ -306,8 +341,22 @@ where
 		nonce: Option<U256>,
 		access_list: Vec<(H160, Vec<H256>)>,
 		is_transactional: bool,
+		weight_limit: Option<Weight>,
+		transaction_len: Option<u64>,
 		evm_config: &evm::Config,
 	) -> Result<(), RunnerError<Self::Error>> {
+		// Try to subtract the transaction_len from the Weight proof_size limit or fail
+		// Validate the weight limit can afford recording transaction len
+		if let (Some(weight_limit), Some(transaction_len)) = (weight_limit, transaction_len) {
+			let _ = weight_limit
+				.proof_size()
+				.checked_sub(transaction_len)
+				.ok_or(RunnerError {
+					error: Error::<T>::GasLimitTooLow,
+					weight: Weight::zero(),
+				})?;
```

Comment:
Consider adding this to the `CheckEvmTransaction`?

Reply:
good idea, done